internal/cloud/realm/templates.go

```diff
@@ -48,6 +50,10 @@ func (c *client) ClientTemplate(groupID, appID, templateID string) (*zip.Reader,
 	if resErr != nil {
 		return nil, resErr
 	}
+	if res.StatusCode == http.StatusNoContent {
+		// No client exists for this template so there is nothing to return
```

Comment:
From my understanding, we return a `204` when there is no `ClientPath` in the template that we're trying to fetch from the `ClientTemplate` endpoint. 
The `ClientPath` is used to get the template content from Github, which is why we need one attached to the template itself. 

So, if there is no `ClientPath`, then we can't get the information for the Template from Github, and therefore, the return for this fetch should be `nil`.

Reply:
I don't think we actually need this check in this function. It should be for the `ClientTemplate` function.

This is referring to the changes made here: https://github.com/10gen/baas/pull/4381/files

Basically, in the process of fetching a client template, sometimes the template that we want to fetch the client for does not have an associated frontend client (like in the case of the database triggers template).

We would return a `http.StatusNoContent` to denote that the template that you're trying to find a client for does not have a template.

Reply:
hahaha i think I thought it was in a different function too 🙄 ... checking for `NoContent` makes more sense to me here.

my only thought is: it leaves us a bit more flexible (and probably more go-like) if we add `bool` to list of return values to this function (before `error`, after `*zip.Reader`) so that we're not relying on a "is nil" check on the returned object to otherwise confirm a valid response.  id rather we have a `zipPkg, ok, err := realmClient.ClientTemplate` type of usage

Reply:
i agree. that's a good idea

Reply:
I forgot to add this, but I will in just a bit

Reply:
hmm i dont see this change for some reason. maybe there was some miscommunication?